server/src/saros/server/console/InviteCommand.java

```diff
@@ -44,9 +51,58 @@ public void execute(List<String> args, PrintStream out) {
         log.warn("Invalid JID skipped: " + jid);
       }
 
-      sessionManager.invite(jids.get(true), "Invitation by server command");
       for (JID jid : jids.get(true)) {
-        sessionManager.startSharingReferencePoints(jid);
+
+        // --------------add contact if not already--------------
+        boolean contactFound = false;
+        for (XMPPContact contact :
+            contactsService.getAllContacts()) { // loop and check if contact was already added
+          if (contact.getBareJid().equals(jid)) {
+            contactFound = true;
+            break;
+          }
+        }
+
+        if (!contactFound) { // add contact
+          log.warn(
+              "Adding "
+                  + jid
+                  + " as new contact. You will have to reinvite him after he accepted being added to the contacts");
+          out.println(
+              "Adding "
+                  + jid
+                  + " as new contact. You will have to reinvite him after he accepted being added to the contacts"); // the previous warning wont be displayed in the console
+
+          final BiPredicate<String, String> questionDialogHandler = // error handler
```

Comment:
It is an automated process, either you are query the STDIN or simply return true, As a side node, this will executed mostly all the time because of privacy reasons, i.e the server will not tell you if such a JID exists.